portal/worker/src/walrus-sites-sw.ts

```diff
@@ -50,8 +48,7 @@ self.addEventListener("fetch", async (event) => {
     console.log("Parsed URL: ", parsedUrl);
 
     if (requestDomain == portalDomain && parsedUrl && parsedUrl.subdomain) {
-        console.log("fetching from the service worker");
-        event.respondWith(resolveAndFetchPage(parsedUrl));
+        await respondUsingCache(event, parsedUrl, urlString);
```

Comment:
Suggestion: for consistency and readability, I would prefer something along these lines:

```suggestion
    if (requestDomain == portalDomain && parsedUrl && parsedUrl.subdomain) {
        if (!("caches" in self)) {
            // When not being in a secure context, the Cache API is not available.
            console.warn("Cache API not available");
            return await resolveAndFetchPage(parsedUrl);
        }
        return await resolveWithCache(parsedUrl);
    }
```

this way it is clear when the cache is used, and we can avoid passing the event.
Further, it may be nice to add the full `urlString` to the `DomainDetails` struct, so that the function signature is similar?

Reply:
I would avoid touching for now the `DomainDetails` type to not affect any server portal functionality.